### DISCUSSION TITLE
persist store to URL in a NextJS application

### DISCUSSION THREAD

**Discussion by ErikMedeiros** [pmndrs/zustand#1755] [Q&A]
Hi! I was messing around with zustand for the first time and wondered how to save a store to the url and read from it, searched a bit and found about the persist middleware but I didn't managed to get it to work.
I tried to copy the example in this [guide](https://docs.pmnd.rs/zustand/guides/connect-to-state-with-url-hash) and messed a little with next/router, both without success.

Any help would be appreciated!

**Comment by JacobWeisenburger** [comment]
> save a store to the url and read from it

What do you mean by this?
